From: Alan Coopersmith <alan.coopersmith@oracle.com>
Subject: [PATCH:tigervnc-trunk] Make fb header transformations work with Xorg 1.2.1

The reformatting of all the Xserver sources in the 1.12.1 release changed
the headers so now some of them have multiple instances of the C++ "and"
and "xor" keywords on the same line, so the hack to sanitize them needs
to use the sed "g" flag to replace all instances, not just the first.

Otherwise, the build fails with strange errors such as:

"fb.h", line 817: Error: Unexpected type name "FbBits" encountered.
"fb.h", line 817: Error: dst is not defined.
"fb.h", line 818: Error: Unexpected type name "FbStride" encountered.
"fb.h", line 818: Error: dstStride is not defined.
"fb.h", line 819: Error: Badly formed expression.

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>

diff --git a/unix/xserver/hw/vnc/Makefile.am b/unix/xserver/hw/vnc/Makefile.am
index 1db0155..6604a87 100644
--- unix/xserver/hw/vnc/Makefile.am
+++ unix/xserver/hw/vnc/Makefile.am
@@ -68,10 +68,10 @@ EXTRA_DIST = Xvnc.man
 BUILT_SOURCES = $(nodist_Xvnc_SOURCES)
 
 fb.h: $(top_srcdir)/fb/fb.h
-	cat $(top_srcdir)/fb/fb.h | sed -e 's,and,c_and,' -e 's,xor,c_xor,' > $(srcdir)/fb.h
+	cat $(top_srcdir)/fb/fb.h | sed -e 's,and,c_and,g' -e 's,xor,c_xor,g' > $(srcdir)/fb.h
 
 pixman.h: $(includedir)/pixman-1/pixman.h
 	cat $(includedir)/pixman-1/pixman.h | sed 's/xor/c_xor/' > $(srcdir)/pixman.h
 
 fbrop.h: $(top_srcdir)/fb/fbrop.h
-	cat $(top_srcdir)/fb/fbrop.h | sed -e 's,and,c_and,' -e 's,xor,c_xor,' > $(srcdir)/fbrop.h
+	cat $(top_srcdir)/fb/fbrop.h | sed -e 's,and,c_and,g' -e 's,xor,c_xor,g' > $(srcdir)/fbrop.h
